common/lib/src/realm_types.dart

```diff
@@ -79,30 +79,140 @@ class RealmStateError extends StateError implements RealmError {
   RealmStateError(String message) : super(message);
 }
 
+/// @nodoc
+class Decimal128 {} // TODO!
+
+/// @nodoc
+class RealmObjectMarker {}
+
+// Union type
+/// @nodoc
+class RealmAny {
+  final dynamic value;
+  T as<T>() => value as T; // better for code completion
+
+  // This is private, so user cannot accidentally construct an invalid instance
+  const RealmAny._(this.value);
+
+  const RealmAny.bool(bool b) : this._(b);
+  const RealmAny.string(String text) : this._(text);
+  const RealmAny.int(int i) : this._(i);
+  const RealmAny.float(Float f) : this._(f);
+  const RealmAny.double(double d) : this._(d);
+  const RealmAny.uint8List(Uint8List data) : this._(data);
+  // TODO: RealmObjectMarker introduced to avoid dependency invertion.
+  // It would be better if we could use RealmObject directly
+  const RealmAny.realmObject(RealmObjectMarker o) : this._(o);
+  const RealmAny.dateTime(DateTime timestamp) : this._(timestamp);
+  const RealmAny.objectId(ObjectId id) : this._(id);
+  const RealmAny.decimal128(Decimal128 decimal) : this._(decimal);
+  const RealmAny.uuid(Uuid uuid) : this._(uuid);
+}
+
+/// Thrown when an error occurs during synchronization
+/// {@category Sync}
+class SyncError extends RealmError {
+  /// The numeric code value indicating the type of the sync error.
+  final int codeValue;
+
+  /// The category of the sync error
+  final SyncErrorCategory category;
+
+  SyncError(String message, this.category, this.codeValue) : super(message);
+
+  /// Creates a specific type of [SyncError] instance based on the [category] and the [code] supplied.
+  static SyncError create(String message, SyncErrorCategory category, int code, {bool isFatal = false}) {
+    switch (category) {
+      case SyncErrorCategory.client:
+        final errorCode = SyncClientErrorCode.values[code];
```

Comment:
values[code] will return "Index out of range" error. We have to find a way to convert codes to enum. Values are working with indexes that are actually the positions in the enum.

Reply:
good catch. Will fix